src/compiler/types.ts

```diff
@@ -1606,6 +1606,16 @@ namespace ts {
         antecedent: FlowNode;
     }
 
+    export type FlowType = Type | IncompleteType;
+
+    // Incomplete types occur during control flow analysis of loops. An IncompleteType
+    // is distinguished from a regular type by a flags value of zero. Incomplete type
+    // objects are internal to the getFlowTypeOfRefecence function and never escape it.
+    export interface IncompleteType {
+        flags: TypeFlags;  // No flags set

```

Comment:
`TypeFlags` isn't a union enum type (because it contains non-literal initializers and is intended to be used as bit flags), so there is no `TypeFlags.None` type. I could use `0`, but then it would be less obvious that `flags` is overlaid on the similarly named property in `Type`. Either way, the comment is still necessary.